core/modules/replica/WorkerProcessor.cc

```diff
@@ -299,9 +298,10 @@ void WorkerProcessor::enqueueForDeletion(string const& id,
         auto const ptr = _requestFactory.createDeleteRequest(
             _worker,
             id,
-            request.priority(),
-            request.database(),
-            request.chunk()
+            priority,
+            bind(&WorkerProcessor::dispose, shared_from_this(), _1),
```

Comment:
would a lambda, even one that just calls `dispose` be more clear here? (I bet you could pass shared_from_this in as an argument?

Reply:
A reason why I chose to avoid a lambda function here would be because the code would look like:
```
auto const self = shared_from_this();
auto const ptr = _requestFactory.createDeleteRequest(
             _worker,
             id,
            request.priority(),
            request.database(),
            request.chunk()
            priority,
            [self] (string const& id) {
                self->dispose(id);
            },
```
I think `bind` is a more concise way of doing it.

Reply:
I find the latter one more readable.  We already have to understand how to read lambdas in this modern world; I would like to be able to dispense with the cognitive load of bind and reclaim those brain cells...

I would argue that putting the bind on one line in such a simple case would also be acceptable.